src/tests/components/FSHControls.test.js

```diff
@@ -242,6 +242,52 @@ it('calls GoFSH with the indent option if the configuration checkbox is checked'
   });
 });
 
+it('displays code with line wrapping in the code editors if the configuration checkbox is checked', async () => {
+  const examplePatient = {
+    resourceType: 'Patient',
+    id: 'MyPatient',
+    gender: 'female'
+  };
+  const simpleFsh = ['Instance: MyPatient', 'InstanceOf: Patient', 'Usage: #example', '* gender = #female'].join('\n');
+  const onGoFSHClick = jest.fn();
+  const resetLogMessages = jest.fn();
+  const setIsLineWrapped = jest.fn();
+  const runGoFSHSpy = jest.spyOn(fshHelpers, 'runGoFSH').mockReset().mockResolvedValue({ fsh: simpleFsh, config: {} });
+  const { getByRole, getByLabelText } = render(
+    <FSHControls
+      onGoFSHClick={onGoFSHClick}
+      gofshText={[{ def: JSON.stringify(examplePatient, null, 2) }]}
+      resetLogMessages={resetLogMessages}
+      exampleConfig={[]}
+      setIsLineWrapped={setIsLineWrapped}
+      isLineWrapped={false}
+    />,
+    container
+  );
+
+  const configButton = getByRole('button', { name: /Configuration/i });
+  fireEvent.click(configButton);
+  const isLineWrappedCheckbox = getByLabelText('Line wrap within code editors');
+  expect(isLineWrappedCheckbox).not.toBeChecked();
+  fireEvent.click(isLineWrappedCheckbox);
+  const button = document.querySelector('[testid=GoFSH-button]');
+  act(() => {
+    button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+  });
+
+  await waitFor(() => {
+    expect(isLineWrappedCheckbox).toBeChecked();
+    expect(resetLogMessages).toHaveBeenCalledTimes(1);
+    expect(runGoFSHSpy).toHaveBeenCalledWith([JSON.stringify(examplePatient, null, 2)], {
+      dependencies: [],
+      indent: false
+    }); // No IG resource added because canonical and version set to defaults
+    expect(onGoFSHClick).toHaveBeenCalledTimes(2);
+    expect(onGoFSHClick).toHaveBeenCalledWith('', true); // Loading
+    expect(onGoFSHClick).toHaveBeenCalledWith(simpleFsh, false);
+  });
```

Comment:
I'm wondering about this test and what we really want to be testing. I don't think we need the checks on lines 280-287 because we don't really need to check the functions that get called as a result of trying to run GoFSH. That probably means we also don't need the actions on lines 273-276 to actually run GoFSH unless we're trying to use that to also check line wrapping in the GoFSH editor, but I don't think this test actually checks that right now.

That pretty much leaves the expect statement on 279 which checks if the checkbox is checked after we click it. I guess it is probably good to test that, so maybe it is worth keeping this test. It might be good to check that when you click the checkbox, that value makes it to the editor component, but that would require changing this test, and maybe not having the test in this file. I'm not sure if we have any tests that check the interaction between the components like that. If that gets complicated though, it might be worth stepping back and considering if that is worth testing.